sqlutil/monitor.go

```diff
@@ -0,0 +1,52 @@
+package sqlutil
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+)
+
+// MonitorDatabase collects database connection pool metrics at regular intervals
+func MonitorDatabase(
+	ctx context.Context,
+	conf *config.Config,
+	statsFactory stats.Stats,
+	db *sql.DB,
+	dbIdentifier string,
+) {
+	reportInterval := conf.GetDurationVar(10, time.Second, "Database.ReportInterval", dbIdentifier+".Database.ReportInterval")
+
+	maxOpenConnectionsStat := statsFactory.NewStat(dbIdentifier+".db.max_open_connections", stats.CountType)
+	openConnectionsStat := statsFactory.NewStat(dbIdentifier+".db.open_connections", stats.CountType)
+	inUseStat := statsFactory.NewStat(dbIdentifier+".db.in_use", stats.CountType)
+	idleStat := statsFactory.NewStat(dbIdentifier+".db.idle", stats.CountType)
+	waitCountStat := statsFactory.NewStat(dbIdentifier+".db.wait_count", stats.CountType)
```

Comment:
wouldn't a gauge make more sense here..?

Reply:
Probably everything can be Gauge instead of `WaitDuration`?

Reply:
yeah